configure.ac

```diff
@@ -733,10 +743,43 @@ AS_IF([test x"$ac_cv_type_cpuset_t" = xyes -o x"$ac_cv_type_cpu_set_t" = xyes ],
 	AC_CHECK_FUNC(cpuset_isset)
 	AC_CHECK_FUNC(cpuset_size)
 	AC_LIBOBJ(cpuset)
-	AC_CHECK_FUNCS([sysconf sched_setaffinity])
+	AC_CHECK_FUNCS([sysconf])
 	AC_CHECK_CPU_OR
 ])
 
+#
+# sched_setaffinity must be checked using proper includes.
+# also needs _GNU_SOURCE on Linux and Hurd; see above.
+# also see https://github.com/NLnetLabs/nsd/issues/82.
+#
+AC_MSG_CHECKING(for sched_setaffinity)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+  [[
+    #ifdef HAVE_SCHED_H
+    # include <sched.h>
+    #endif
+    #ifdef HAVE_SYS_CPUSET_H
+    #include <sys/cpuset.h>
+    #endif
+    #ifdef HAVE_CPUSET_T
+    #define MY_CPUSET_TYPE cpuset_t
+    #endif
+    #ifdef HAVE_CPU_SET_T
+    #define MY_CPUSET_TYPE cpu_set_t
+    #endif
+    #ifdef HAVE_CPUID_T
+    #define MY_CPUSET_TYPE cpuid_t
+    #endif
```

Comment:
Delete these thee lines, the ifdef HAVE_CPUID_T.  That is for the item in the set, not the set type.

Reply:
Fixed in https://github.com/NLnetLabs/nsd/commit/66383945d13eed7c875570956daddaadae116f6e .  Also for the CPU_OR test.

Reply:
Perfect, thanks.